### PR TITLE
Account for new changes in ggplot2's internal API, fixes #1561

### LINKED ISSUE #1561: ggplotly broken under lastest ggplot2 dev version?
Seems that ggplot2 dev version 3.2.0.9000 has broken something in ggplotly.

First, with ggplot2 stable version (the result is as expected), the names of the labels of x-axis.
Second, with ggplot2 dev version (the result is NA), no names of labels.

Both tests made with lastest stable plotly version 4.9.0 under lastest R-devel for Windows 10.
---

Using stable version of ggplot2 3.2.0

```r
library(plotly)
#> Loading required package: ggplot2
#> 
#> Attaching package: 'plotly'
#> The following object is masked from 'package:ggplot2':
#> 
#>     last_plot
#> The following object is masked from 'package:stats':
#> 
#>     filter
#> The following object is masked from 'package:graphics':
#> 
#>     layout
p <- ggplot(iris) +
  geom_point(aes(x = Petal.Width, y = Sepal.Length, color = Species)) +
  scale_x_continuous(breaks = seq(0,3,0.5), limits =c(0,3), labels = letters[1:7])
gg <- ggplotly(p)
gg$x$layout$xaxis$ticktext
#> [1] "a" "b" "c" "d" "e" "f" "g"
```
Using the lastest dev version of ggplot2:

```r
library(plotly)
#> Loading required package: ggplot2
#> 
#> Attaching package: 'plotly'
#> The following object is masked from 'package:ggplot2':
#> 
#>     last_plot
#> The following object is masked from 'package:stats':
#> 
#>     filter
#> The following object is masked from 'package:graphics':
#> 
#>     layout
p <- ggplot(iris) +
  geom_point(aes(x = Petal.Width, y = Sepal.Length, color = Species)) +
  scale_x_continuous(breaks = seq(0,3,0.5), limits =c(0,3), labels = letters[1:7])
gg <- ggplotly(p)
gg$x$layout$xaxis$ticktext
#> [1] NA
```

### PR DESCRIPTION
**Pull request by cpsievert**
See #1561 and https://github.com/tidyverse/ggplot2/issues/3436

### CODE REVIEW

File: R/ggplotly.R [hunk 1]
```diff
@@ -492,10 +492,10 @@ gg2list <- function(p, width = NULL, height = NULL,
   layout$layout$xanchor <- paste0("y", sub("^1$", "", layout$layout$xanchor))
   layout$layout$yanchor <- paste0("x", sub("^1$", "", layout$layout$yanchor))
   # for some layers2traces computations, we need the range of each panel
-  layout$layout$x_min <- sapply(layout$panel_params, function(z) min(z$x.range %||% z$x_range))
-  layout$layout$x_max <- sapply(layout$panel_params, function(z) max(z$x.range %||% z$x_range))
-  layout$layout$y_min <- sapply(layout$panel_params, function(z) min(z$y.range %||% z$y_range))
-  layout$layout$y_max <- sapply(layout$panel_params, function(z) max(z$y.range %||% z$y_range))
+  layout$layout$x_min <- sapply(layout$panel_params, function(z) min(z[["x"]]$dimension %()% z$x.range %||% z$x_range))
```
Comment (schloerke) [inline R/ggplotly.R]: what's order of operations on this?

`a %()% (b %||% c)` ?

File: R/ggplotly.R [hunk 2]
```diff
@@ -552,7 +552,7 @@ gg2list <- function(p, width = NULL, height = NULL,
       )
       # allocate enough space for the _longest_ text label
       axisTextX <- theme[["axis.text.x"]] %||% theme[["axis.text"]]
-      labz <- unlist(lapply(layout$panel_params, "[[", "x.labels"))
+      labz <- unlist(lapply(layout$panel_params, function(pp) pp[["x"]]$get_labels %()% pp$x.labels))
```
Comment (schloerke) [inline R/ggplotly.R]: use `{``}`

Comment (cpsievert) [reply R/ggplotly.R]: Why, just for style?

File: R/ggplotly.R (hunk 1 above)
Comment (cpsievert) [reply R/ggplotly.R]: `(` takes precedence over infix operators (e.g., `%||%`, `%()%`)

**Comment by cpsievert** [comment]
> `%()%` seems scary as I don't know what it's actually grabbing (unless it behaves exactly like `%>%`).

Can you elaborate? It behaves very similarly to `%||%` (not `%>%`).

Can you propose an alternative? This is going to be a common enough pattern (if a certain field is a function, call it, otherwise fallback on something else) that I'd like a helper for it

File: R/ggplotly.R (hunk 1 above)
Comment (cpsievert) [reply R/ggplotly.R]: (BTW, I'm pretty sure infix operators have the same precendence, so you can read them left to right, unless you're using `(` for some reason)

File: R/ggplotly.R (hunk 2 above)
Comment (schloerke) [reply R/ggplotly.R]: is it `(function(x) x) %()% y` or `function(x) { x %()% y}` ?

Hard to read. So I guess style.

**Pull request merged**